### PR TITLE
Add longhorn 1.2.0 chart for cluster manager

### PR DESCRIPTION
**Pull request by innobead**
Signed-off-by: David Ko <dko@suse.com>

This PR is ready for review, but the images will be finalized until the GA release of v1.2.0, so it is marked as a draft for now.

cc @khushboo-rancher @meldafrawi

### CODE REVIEW

**Comment by khushboo-rancher** [comment]
Validated with the branch `add-longhorn-1.2.0-chart-master`

Validation - **Pass**

**Comment by aiyengar2** [comment]
@innobead we should close out this PR since it's to the wrong branch

**Comment by innobead** [comment]
> @innobead we should close out this PR since it's to the wrong branch

Which branch is the correct one?

File: charts/longhorn/latest/Chart.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 apiVersion: v1
 name: longhorn
-version: 1.1.2
-appVersion: v1.1.2
-kubeVersion: ">=v1.16.0-r0"
+version: 1.2.0
+appVersion: v1.2.0
+kubeVersion: ">=v1.18.0"
```
Comment (shuo-wu) [inline charts/longhorn/latest/Chart.yaml:5]: May need to change this to `"kubeVersion: >=1.18.0-r0"`

See https://github.com/longhorn/longhorn/issues/2963

**Comment by aiyengar2** [comment]
> @innobead we should close out this PR since it's to the wrong branch

Sorry for the confusion; just checked in with @cbron about this and this is the right branch.

**Comment by innobead** [comment]
Before merging, we need QA to verify again before merging. cc @khushboo-rancher

**Review by aiyengar2 — COMMENTED** [review]
One clarification: why does longhorn place the latest chart under `latest/` instead of under `1.2.0`? Seems like longhorn is the only chart that does this.

Otherwise, these changes LGTM

**Comment by cbron** [comment]
@innobead are we going to wait to merge until its equivalent in cluster-explorer goes out with rancher release ? To avoid situation we hit last time about users asking when its going live.

**Review by jiaqiluo — APPROVED** [review]

**Review by shuo-wu — APPROVED** [review]

**Comment by innobead** [comment]
> @innobead are we going to wait to merge until its equivalent in cluster-explorer goes out with rancher release ? To avoid situation we hit last time about users asking when its going live.

We don't need to wait to merge, because we don't know how long it will take and at least let users using cluster manager able to install/upgrade longhorn earlier. 

As you mentioned, I believe people will ask a similar question why 1.2.0 is still unavailable in cluster explorer, but you already have a good explanation for 1.1.2 about the difference between cluster manager and cluster explore in https://github.com/rancher/charts/issues/1352#issuecomment-887999551, so we can probably share this with users if they have the question.

**Comment by innobead** [comment]
@khushboo-rancher we did a few updates for this PR, so please help verify again to ensure the PR working as expected. Thank you!

**Comment by meldafrawi** [comment]
I've validated the chart again, **LGTM**.

**Review by jiaqiluo — APPROVED** [review]
LGTM

**Comment by khushboo-rancher** [comment]
https://github.com/longhorn/longhorn/issues/2964 is fixed with this PR

**Review by aiyengar2 — APPROVED** [review]
Already gave an LGTM but re-approving since it got dismissed.

@innobead @shuo-wu @cbron I think we still need to clarify the following:

> One clarification: why does longhorn place the latest chart under `latest/` instead of under `1.2.0/`? Seems like longhorn is the only chart that does this.

**Comment by yasker** [comment]
@aiyengar2 It was required to do so when we started publishing the chart a few years back. I guess somewhere along the way Rancher has changed the chart layout? I found at least it was still like this in late 2019. https://github.com/rancher/charts/tree/325754b5123b6e3358504c72a453dad2aea39d82 

It's much better if we can skip the `latest` folder.

**Comment by aiyengar2** [comment]
Thanks @yasker! Yes I don't think it's a blocker. 

w.r.t. the fix, I'm also worried that a fix will introduce an error into Rancher due to some code dependency that may have necessitated it in the first place (which is why I asked about the context). But since this branch is practically deprecated in Rancher 2.6.0 (since Cluster Manager is no longer supported), I think it's fine to leave it as-is.

**Comment by yasker** [comment]
Thanks @aiyengar2 .

I noticed the CI failed. Is it a blocker?

**Comment by yasker** [comment]
@cbron Can you take a final look? Thanks!

**Review by cbron — APPROVED** [review]
Branches lgtm, not reviewing content since its already approved by longhorn team.

**Pull request merged**